lib/LruMemoize.js

```diff
@@ -3,11 +3,32 @@
  */
 import LRU from 'lru-cache';
 
+/**
+ * LruMemoize uses the npm module `lru-cache` to memoize async functions.
+ *
+ * @see https://www.npmjs.com/package/lru-cache
+ * @see https://en.wikipedia.org/wiki/Memoization
+ * @param {object} cacheOptions - Options for `lru-cache`.
+ *   See the npm docs for more options.
+ * @param {number} [cacheOptions.max] - The max size of the cache.
+ * @param {number} [cacheOptions.maxAge] - The maxAge of an item in ms.
+ *
+ * @returns {LruMemoize} The class.
+*/
 export default class LruMemoize {
   constructor(cacheOptions) {
     this.cache = new LRU(cacheOptions);
   }
 
+  /**
+   * Memoizes an async function in the cache.
+   *
+   * @param {object} options - Options to use.
+   * @param {string} options.key - A key for the cache.
+   * @param {Function<Promise>} options.fn - An Async Function to memoize.
```

Comment:
@dlongley how do you feel about this language.  I'm thinking something more like:
`A function that returns an unresolved promise.`

Reply:
Went with Longley's wording here.